### ISSUE TITLE
Avoid computing `FormFeature` when calling `HttpRequest.HasFormContentType`

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#50153]
Currently, invoking the `HttpRequest.HasFormContentType` property will implicitly compute a `FormFeature` and use use that to compute whether the content type is form-related:

https://github.com/dotnet/aspnetcore/blob/76d2fda2659928859972211d99b5457180f2a8d8/src/Http/Http/src/Internal/DefaultHttpRequest.cs#L150-L153

We should consider updating this implementation such that it checks the content-type directly instead of allocating a feature.

cc: @davidfowl @Tratcher

**Comment by Tratcher** [comment]
Just make sure the checks are consistent/equivalent. E.g. the alternative check you added in another recent PR was simplified for perf reasons and won't always return the same result.

**Comment by captainsafia** [comment]
> Just make sure the checks are consistent/equivalent. E.g. the alternative check you added in another recent PR was simplified for perf reasons and won't always return the same result.

Correct. I'd personally be fine with just calling [these methods](https://source.dot.net/#Microsoft.AspNetCore.Http/Features/FormFeature.cs,306) directly as a step towards optimization here.

**Comment by davidfowl** [comment]
I think we'd want to do a couple of investigations:
- Avoid allocating the MediaTypeHeader (see https://github.com/dotnet/aspnetcore/blob/376b6b0c860f113d59cf3e82677d7f8a6e674716/src/Mvc/Mvc.Core/src/Formatters/MediaType.cs)
- Avoid allocating the form feature if there isn't one already.
- This antiforgery check seems to happen when reading the form (https://github.com/dotnet/aspnetcore/blob/376b6b0c860f113d59cf3e82677d7f8a6e674716/src/Http/Http/src/Features/FormFeature.cs#L77), which happens when you check the content type. This seems wrong.
